app/models/grda_warehouse/hud/client.rb

```diff
@@ -2391,28 +2400,41 @@ def merge_from(other_client, reviewed_by:, reviewed_at:, client_match_id: nil)
     end
 
     def move_dependent_hmis_items(previous_id, new_id)
+      return if previous_id == new_id
+
       hmis_dependent_items.each do |klass|
         klass.where(client_id: previous_id).
           update_all(client_id: new_id)
       end
     end
 
     def move_dependent_health_items(previous_id, new_id)
-      health_dependent_items.each do |klass|
-        klass.where(client_id: previous_id).
-          update_all(client_id: new_id)
+      return if previous_id == new_id
+
+      # If we are merging 2 existing patients...
+      previous_patient = Health::Patient.find_by(client_id: previous_id)
+      new_patient = Health::Patient.find_by(client_id: new_id)
+      if previous_patient.present? && new_patient.present?
+        # Confirm their MedicaidIDs match
+        raise Health::MedicaidIdConflict, "Cannot merge #{previous_patient.id} and #{new_patient.id}" if previous_patient.medicaid_id != new_patient.medicaid_id
```

Comment:
Can we `ping` this and return instead of raising?

Reply:
By raising, it can be caught in the enclosing transaction scope, and I was thinking the merge should be all or nothing.

Reply:
wouldn't raising prevent any other pending merges from happening?  Maybe we need a rescue around the transaction so that flow can continue even if one client merge fails.